### PR TITLE
📝 Add Opencontainers labels in docker image

### LINKED ISSUE #1772: 📝 Use opencontainers labels in docker image
:memo: Use opencontainers labels in docker image

[label-schema.org](https://github.com/label-schema/label-schema.org#label-schemaorg) has officially announced that their specification is deprecated in favor of the [OCI annotations](https://github.com/opencontainers/image-spec/blob/master/annotations.md#back-compatibility-with-label-schema).

> the OCI has defined a parallel set of similar conventions in the OCI Image Spec (known as annotations rather than labels) it makes sense to consolidate efforts in one place.
> This project is no longer actively maintained. Visit http://label-schema.org for more information about the legacy label-schema project, and go to the OCI Image Spec for future label definitions.

### PR DESCRIPTION
**Pull request by madmath03**
:memo: Add opencontainers labels in docker image to fix #1772

Not removing label-schema labels (for now at least), but just adding the opencontainers equivalent labels.

Signed-off-by: mathieu.brunot <mathieu.brunot@monogramm.io>

### CODE REVIEW

**Comment by return42** [comment]
Sorry for my off topic, nitpicking question .. 

Why did you tag the commit label ``:memo:``, I think  ``docker:`` matches better, I am wrong?

Are there any recommended comment-styles out there? .. I found [Git Commit Good Practice](https://wiki.openstack.org/wiki/GitCommitMessages#Information_in_commit_messages)

**Comment by madmath03** [comment]
Hi @return42,

> Sorry for my off topic, nitpicking question ..

No problem, I don't mind being picky 😄 

> Why did you tag the commit label :memo:, I think docker: matches better, I am wrong?

Well, you're not wrong, as this is indeed part of the docker container, but those labels have no actual impact on the docker image. They are just here to document the image built... so I guess both :memo: `:memo:` and :whale: `:whale:` or `docker:` would be valid.

> Are there any recommended comment-styles out there? .. I found Git Commit Good Practice

Well, there isn't one universal way, more like a shared consensus that commit messages should have a type and short description, but the details and format vary from project to project.

[Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) is a good reference. There even is a GitHub bot to enforce PR / commits to follow this format if that's what you want: [Semantic Pull Requests](https://github.com/zeke/semantic-pull-requests)

Personally, I prefer to use [gitmoji](https://gitmoji.carloscuesta.me/) which has pretty much the same idea but just uses emojis for prettier / quicker reading. There is no GitHub bot (yet) to enforce this.

Apart from GitHub bots, maybe this could be added to the contributing guides if you wish to enforce some rules on this ?

**Comment by return42** [comment]
Hi @madmath03 ,

thanks for your helpful feedback!

> Apart from GitHub bots, maybe this could be added to the contributing guides if you wish to enforce some rules on this?

I added some comments in the "Code" section #1776, see my preview (don't bookmark): https://return42.github.io/searx/dev/contribution_guide.html#code

TL;DR; 

We should not over engineer the commit messages, but we need some simple roles to keep quality.  More over, it is much more important [to split changes (patches) structural](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes).  I added such know-how (refs) to the contrib-guide in the hope it is used meaningful and gives new contributors a orientation.

At the end I think, we should not define to much hard & strict rules, instead we should instruct and support our contributors to work properly.

nitpicking  ..

>> I think docker: matches better, I am wrong?

> Well, you're not wrong, as this is indeed part of the docker container, but those labels have no actual impact on the docker image. They are just here to document the image built... so I guess both memo :memo: and whale :whale: or docker: would be valid.

From my  very personal opinion, `docker(labels):` fits more .. 

.. but don't change it! I just picked this commit up to discuss and hear what other think about commit messages and structural split of the patches.

**Comment by return42** [comment]
@dalf can you pick up this PR?

**Review by dalf — APPROVED** [review]
Sorry for the delay.
